azure/scope/managedcontrolplane.go

```diff
@@ -328,7 +341,18 @@ func (s *ManagedControlPlaneScope) IsIPv6Enabled() bool {
 
 // IsVnetManaged returns true if the vnet is managed.
 func (s *ManagedControlPlaneScope) IsVnetManaged() bool {
-	return true
+	if s.cache.isVnetManaged != nil {
+		return to.Bool(s.cache.isVnetManaged)
+	}
+	ctx := context.Background()
+	ctx, log, done := tele.StartSpanWithLogger(ctx, "scope.ManagedControlPlaneScope.IsVnetManaged")
+	defer done()
+	isManaged, err := virtualnetworks.New(s).IsManaged(ctx)
```

Comment:
In `cluster.go` you called this variable `isVnetManaged`.

Reply:
There's nothing mysterious here, I'm just preferring to use a lowercase local var to capture the response from the method name that we're calling.